holoviews/plotting/mpl/plot.py

```diff
@@ -82,8 +82,7 @@ class MPLPlot(DimensionedPlot):
     sublabel_size = param.Number(default=18, doc="""
          Size of optional subfigure label.""")
 
-    projection = param.ObjectSelector(default=None,
-                                      objects=['3d', 'polar', None], doc="""
+    projection = param.Parameter(default=None, doc="""
```

Comment:
Doesn't validate yet but could do, although some matplotlib extensions allow new projection types to be specified as strings.